signbank/dictionary/adminviews.py

```diff
@@ -6713,8 +6703,9 @@ def get_queryset(self):
             # the query set is a list of tuples (gloss, keyword_translations, senses_groups)
             return []
 
-        if ('search_type' in self.request.session.keys() and
-                self.request.session['search_type'] != self.search_type):
+        session_search_type = self.request.session['search_type'] if 'search_type' in self.request.session.keys() else ''
```

Comment:
You could do something like `... = self.request.session.get('search_type', '')` here. See also https://www.w3schools.com/python/ref_dictionary_get.asp